typings/index.d.ts

```diff
@@ -2363,11 +2363,11 @@ declare module 'discord.js' {
 
 	type OverwriteType = 'member' | 'role';
 
-	interface PermissionFlags extends Record<PermissionString, number> { }
+	type PermissionFlags = Record<PermissionString, number>;
```

Comment:
Any reason for changing these from interfaces to types for the sake of changing?

Reply:
Interfaces and Types are nigh interchangeable and especially in this case. Using an Interface here however seemed pointless to me as PermissionFlags adds nothing to what it extends from. Now if it would itself have additional properties then yeah an interface with extends is nice but otherwise there really is no need to imho. In my own experience writing types is nearly always shorter and I try to only use Interfaces for class implementations (`class X implements Y`)

Reply:
To add: It's a bit of a mess to use interfaces and types mingled. Linters such as tslint (which is actually used by the project for typings) for that reason have rules to prefer the one over the other, but no such rule has been enabled.